File: pig_project/lib/model/MoneyData.dart
class MoneyData {
  final int amount;
  final String category;
  final String paymentMethod;
  final String content;
  final String type;
  final String date;

  MoneyData({
    required this.amount,
    required this.category,
    required this.paymentMethod,
    required this.content,
    required this.type,
    required this.date,
  });

  factory MoneyData.fromJson(Map<String, dynamic> json) {
    return MoneyData(
      amount: json['amount'] ?? 0, // null 값인 경우 기본값 0 사용
      category: json['category'] ?? 'N/A', // null 값인 경우 기본값 "N/A" 사용
      paymentMethod: json['paymentMethod'] ?? 'N/A', // null 값인 경우 기본값 "N/A" 사용
      content: json['content'] ?? 'N/A', // null 값인 경우 기본값 "N/A" 사용
      type: json['type'] ?? 'N/A', // null 값인 경우 기본값 "N/A" 사용
      date: json['date'] ?? '', // null 값인 경우 기본값 빈 문자열 사용
    );
  }

   Map<String, dynamic> toJson() {
    return {
      'amount': amount,
      'category': category,
      'paymentMethod': paymentMethod,
      'content': content,
      'type': type,
      'date': date,
    };
  }

  
}

File: goldenPig/src/main/java/com/yeop/goldenPig/vo/MoneyData.java
package com.yeop.goldenPig.vo;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@Table(name = "MONEYDATA")
public class MoneyData {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long OID;

    @Column(name = "ANDROIDID", nullable = false )
    private String androidID;

    @Column(name = "DATE", nullable = false)
    private LocalDateTime date;

    @Column(name = "AMOUNT", nullable = false)
    private int amount;

    @Column(name = "CATEGORY", nullable = false)
    private String category;

    @Column(name = "PAYMENTMETHOD", nullable = false)
    private String paymentMethod;

    @Column(name = "CONTENT", nullable = false)
    private String content;

    @Column(name = "TYPE", nullable = false)
    private String type;
}
